presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -0,0 +1,227 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.metastore.glue;
+
+import com.amazonaws.services.glue.AWSGlueAsync;
+import com.amazonaws.services.glue.model.ColumnStatistics;
+import com.amazonaws.services.glue.model.DeleteColumnStatisticsForPartitionRequest;
+import com.amazonaws.services.glue.model.DeleteColumnStatisticsForTableRequest;
+import com.amazonaws.services.glue.model.GetColumnStatisticsForPartitionRequest;
+import com.amazonaws.services.glue.model.GetColumnStatisticsForPartitionResult;
+import com.amazonaws.services.glue.model.GetColumnStatisticsForTableRequest;
+import com.amazonaws.services.glue.model.GetColumnStatisticsForTableResult;
+import com.amazonaws.services.glue.model.UpdateColumnStatisticsForPartitionRequest;
+import com.amazonaws.services.glue.model.UpdateColumnStatisticsForTableRequest;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import io.airlift.concurrent.MoreFutures;
+import io.prestosql.plugin.hive.HiveBasicStatistics;
+import io.prestosql.plugin.hive.metastore.Column;
+import io.prestosql.plugin.hive.metastore.HiveColumnStatistics;
+import io.prestosql.plugin.hive.metastore.Partition;
+import io.prestosql.plugin.hive.metastore.Table;
+import io.prestosql.plugin.hive.metastore.glue.converter.GlueStatConverter;
+import io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.statistics.ColumnStatisticType;
+import io.prestosql.spi.type.Type;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil.getHiveBasicStatistics;
+
+public class DefaultGlueColumnStatisticsProvider
+        implements GlueColumnStatisticsProvider
+{
+    private static final int GLUE_COLUMN_READ_STAT_PAGE_SIZE = 100;
+    private static final int GLUE_COLUMN_WRITE_STAT_PAGE_SIZE = 25;
+
+    private final AWSGlueAsync glueClient;
+    private final String catalogId;
+    private final Executor readExecutor;
+    private final Executor writeExecutor;
+
+    public DefaultGlueColumnStatisticsProvider(AWSGlueAsync glueClient, String catalogId, Executor readExecutor, Executor writeExecutor)
+    {
+        this.glueClient = glueClient;
+        this.catalogId = catalogId;
+        this.readExecutor = readExecutor;
+        this.writeExecutor = writeExecutor;
+    }
+
+    @Override
+    public Set<ColumnStatisticType> getSupportedColumnStatistics(Type type)
+    {
+        return ThriftMetastoreUtil.getSupportedColumnStatistics(type);
```

Comment:
Nevermind, I saw that you are picking the impl. based on the config toggle so that impls. don't need to be aware if things are enabled or disabled. Nice.

Reply:
Yes, I'm not sure if having a No op implementation is a common pattern in the codebase, we may also change the default implementation to use a toggle flag if you think it's more appropriate

Reply:
We do have some no-op implementations in other places in the code. It looks fine to me.

But I'll let the maintainers take the call on this one.

cc: @findepi @electrum

Reply:
We did have a no-op implementation because stats for glue was extensions point. It was needed until Glue provided first-class stats support.
We should maintain the extension point for a while, before we can remove it.
So the DefaultGCSP can change its' behavior, but should not be used uncondintionally. cc @sopel39 

Also, we need a safety toggle to be able to ignore stats, should users run into any problems.
@GaruGaru can you please confirm `hive.table-statistics-enabled` can be used to skip getting stats from Glue?

Reply:
yes disabling `hive.table-statistics-enabled` will use the current No op implementation

Reply:
In order to enable/disable the statistics get/update I've introduced the `hive.metastore.glue.statistics-enabled` flag, do you think we can use `hive.table-statistics-enabled` directly or we should check them both ?

Reply:
One flag should be enough. You cannot use 2 metastores at the same time anyway.

Reply:
I've removed `hive.metastore.glue.statistics-enabled` in favour of `hive.table-statistics-enabled`